admin/users/views.py

```diff
@@ -90,21 +91,20 @@ def delete(self, request, *args, **kwargs):
 
     def get_context_data(self, **kwargs):
         context = {}
-        context.setdefault('guid', kwargs.get('object').pk)
+        context.setdefault('guid', kwargs.get('object')._id)
         return super(UserDeleteView, self).get_context_data(**context)
 
     def get_object(self, queryset=None):
         return OSFUser.load(self.kwargs.get('guid'))
 
 
-class SpamUserDeleteView(UserDeleteView, PermissionRequiredMixin):
+class SpamUserDeleteView(UserDeleteView):
```

Comment:
It inherets from the `UserDeleteView`, which has the `PermissionRequiredMixin` and requires the `osf.change_user` permission already. Though I do suppose this view in particular should also require the `osf.delete_user` permission as well.